libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.ts

```diff
@@ -33,17 +47,55 @@ export class FavoriteStarComponent {
   record_: MetadataRecord
   favoriteCount: number | null
   loading = false
+  DEFAULT_GN4_LOGIN_URL = `/geonetwork/srv/${
+    LANG_2_TO_3_MAPPER[this.translateService.currentLang]
+  }/catalog.signin?redirect=`
+  loginUrl = `${
+    this.optionalLoginUrl ? this.optionalLoginUrl : this.DEFAULT_GN4_LOGIN_URL
+  }${window.location}`
```

Comment:
This could go in AuthService maybe? this feels like too much responsibility for this component alone

Reply:
Also I think simply appending the `window.location` value to the login url here is too "magical". What we need to do is clearly offer to administrators the possibility to put a token inside the url in the app config, such as:

`/cas/login?service=${current_url}` (similar to tokens available in the external viewer settings)

Also the default value should be:

`/geonetwork/srv/${lang3}/catalog.signin?redirect=${current_url}`

so we also need to evaluate this token. Tokens should be evaluated on request, with something like that on AuthService:

```
get loginUrl() {
  return this.baseLoginUrl.replaceAll('${current_url}', window.location.toString()).replaceAll('${lang3}', ...)
}
```